aten/src/ATen/NativeFunctions.h

```diff
@@ -112,8 +112,12 @@ type_method_definition_dispatch: at::native::expand
 */
 static inline Tensor expand(const Tensor &self, IntList size) {
   if (size.size() < (size_t)self.dim()) {
-    throw std::runtime_error("the number of sizes provided must be greater or equal to the "
-                             "number of dimensions in the tensor");
+    std::ostringstream ss;
+    ss << "at::expand(" << self.type() << "{" << self.sizes() << "}, size=" << size
```

Comment:
Can we not mention any C++ related things in here? These are messages visible to Python users and they don't know what `at::expand` is. Just say what's the problem

Reply:
So, we have a more general problem, where the new ATen code may raise an exception somewhere deep inside an implementation, even though the user visible trace doesn't mention the problem at all. For example, I triggered this error message above by passing an incorrectly shaped gradient out of backwards. Then I got an "expand" error when invoking "backward()". Very perplexing. Inspecting the C++ trace in gdb helped elucidate the situation, but this is not very user friendly.

Reply:
I understand that you can raise a C++ specific error when there's e.g. some ATen invariant that was violated, but in this case there is absolutely no need to mention ATen. It only makes it more confusing for people who don't know that it exists